Scribe/ChangeHistory/ChangeManager.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+
+namespace Scribe.ChangeHistory
+{
+    /// <summary>
+    /// Tracks requests to make changes and stores them as <see cref="Change"/> objects to enable Undo and Redo functionality.
+    /// </summary>
+    internal static class ChangeManager
+    {
+        /// <summary>A history of actions taken.</summary>
+        private static List<Change> Changes = new List<Change>();
+
+        /// <summary>The index of the <see cref="Change"/> that will be Undone next.</summary>
+        private static int CurrentChangeIndex = -1;
+
+        /// <summary>The backing field for <see cref="MaximumChanges"/>.</summary>
+        private static int _maximumChanges = 100;
+
+        /// <summary>
+        /// How many levels of undo to maintain.
+        /// No more than this number of <see cref="Change"/>s will ever be stored.  Cannot be less than 1.
+        /// </summary>
+        internal static int MaximumChanges
+        {
+            get => _maximumChanges;
+            set
+            {
+                if (value > 0)
+                {
+                    _maximumChanges = value;
+                }
+                if (Changes.Count > MaximumChanges)
+                {
+                    Changes = Changes.GetRange(Changes.Count - MaximumChanges, MaximumChanges);
+                    CurrentChangeIndex = Changes.Count - 1;
```

Comment:
nit: Imagine someone takes actions, in order, ABCDEF. They undo, and now the pointer is on AB_**C**_DEF. The Maximum Changes value changes from 6 to 5. Their pointer would be moved to ABCD**_E_**, causing errors.